modules/internal/ChapelArray.chpl

```diff
@@ -2358,15 +2360,25 @@ module ChapelArray {
 
     */
     pragma "fn returns aliasing array"
-    proc reindex(d: domain)
-      where isRectangularDom(this.domain) && isRectangularDom(d)
+    inline proc reindex(newDomain: domain)
+      where isRectangularDom(this.domain) && isRectangularDom(newDomain)
+    return reindex((...newDomain.dims()));
+
+    pragma "no doc"
```

Comment:
I'd like to remove this "no doc" pragma so that this function signature shows up in the documentation.  I'd also like to add a bit of chpldoc text for this function.